hdk/virtual_machine/code.py

```diff
@@ -0,0 +1,167 @@
+"""Provides functions to translate VM commands into their assembly commands."""
+from collections.abc import Iterable, Iterator
+
+from hdk.virtual_machine.syntax import (
+    ArithmeticLogicalCommand,
+    MemoryTransferCommand,
+    VMCommand,
+)
+
+_BINARY_TABLE = {
+    "add": "M=D+M",
+    "sub": "M=M-D",
+    "and": "M=D&M",
+    "or": "M=D|M",
+}
+
+_COMPARISON_TABLE = {
+    "eq": "JEQ",
+    "lt": "JLT",
+    "gt": "JGT",
+}
+
+_SEGMENT_TABLE = {
+    "local": "LCL",
+    "argument": "ARG",
+    "this": "THIS",
+    "that": "THAT",
+}
+
+_SAVE_D_INSTRUCTIONS = ["@SP", "A=M", "M=D", "@SP", "M=M+1"]
+_RESTORE_D_INSTRUCTIONS = ["@SP", "AM=M-1", "D=M", "@R13", "A=M", "M=D"]
+
+
+def translate_arithmetic_logical_command(
+    command: ArithmeticLogicalCommand, command_id: int
+) -> list[str]:
+    """Translates an ArithmeticLogicalCommand into its assembly code.
+
+    Args:
+        command (StackInstruction): The stack instruction to translate.
```

Comment:
Remove information in brackets. The docstring style is the following:
Args:
       arg_name: Arg description.

Also change the descriptions to reflect the names of arguments.